sdk/tools/openrewrite-sample-compiler-ci.yml

```diff
@@ -0,0 +1,73 @@
+trigger:
+  branches:
+    include:
+      - main
+  paths:
+    include:
+      - /sdk/tools/
+
+extends:
+  template: /eng/pipelines/templates/stages/1es-redirect.yml
+  parameters:
+    stages: 
+      - stage: CompileSamples
+        variables: 
+          - template: /eng/pipelines/templates/variables/globals.yml
+          - template: /eng/pipelines/templates/variables/image.yml
+
+        jobs:
+          - job: OpenRewriteCompilerPlugin
+            pool: 
+              name: $(LINUXPOOL)
+              image: $(LINUXVMIMAGE)
+              os: linux
+            displayName: Build and Run OpenRewrite Compiler Plugin
+            timeoutInMinutes: 60
+            steps:
+              - template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
+                parameters:
+                  Paths:
+                    - /sdk/tools/azure-openrewrite-compiler-maven-plugin/
+                    - /sdk/tools/azure-openrewrite/
+                    - /sdk/core-v2/
+                    - sdk/clientcore/
+                    - sdk/parents/
+                    - eng/
+                    - common/perf-test-core/
+
+              - task: PowerShell@2
+                displayName: 'Install Required Dependencies from Source'
+                inputs:
+                  targetType: 'inline'
+                  script: |
+                    $modulePaths = @(
+                      "eng/code-quality-reports/pom.xml",
+                      "sdk/parents/pom.xml",
+                      "common/perf-test-core/pom.xml",
+                      "sdk/clientcore/pom.xml",
+                      "sdk/core-v2/pom.xml"
+                    )
+                    foreach ($modulePath in $modulePaths) {
+                      mvn -f $modulePath clean install --% -B -Dcheckstyle.skip -Dgpg.skip -Dmaven.javadoc.skip -Drevapi.skip -DskipSpringITs -DskipTests -Dspotbugs.skip -Djacoco.skip -Dspotless.skip
```

Comment:
If we run `mvn install -pl com.azure.v2:azure-core -am`, it should build all dependencies. `-am` is for building all the dependencies.